numpy/f2py/f2py2e.py

```diff
@@ -615,7 +630,7 @@ def run_compile():
         sysinfo_flags = [f[7:] for f in sysinfo_flags]
 
     _reg2 = re.compile(
-        r'--((no-|)(wrap-functions|lower)|debug-capi|quiet|skip-empty-wrappers)|-include')
+        r'--((no-|)(wrap-functions|lower|freethreading-compatible)|debug-capi|quiet|skip-empty-wrappers)|-include')
```

Comment:
because of the argv munging that happens in this function, I also need to modify this regex, in addition to your suggestion @HaoZeke. Some of the tests only do something interesting on the free-threaded python build, which I'm guessing you don't have to test with.

Thanks for your suggestion, it didn't occur to me to just make `get_includes` a more generic interface.